src/views/Signup.vue

```diff
@@ -0,0 +1,206 @@
+                <span class="icon is-small is-left">
+                  <fa-icon icon="user" />
+                </span>
+              </p>
+            </div>
+            <div class="field mb-2">
+              <p class="control has-icons-left">
+                <input
+                  v-model="password"
+                  class="input"
+                  id="password"
+                  type="password"
+                  :placeholder="$gettext('Password')"
+                />
+                <span class="icon is-small is-left">
+                  <fa-icon icon="lock" />
+                </span>
+              </p>
+            </div>
+            <div class="field mb-2">
+              <p class="control has-icons-left">
+                <input
+                  v-model="confirmPassword"
+                  class="input"
+                  id="confirm-password"
+                  type="password"
+                  :placeholder="$gettext('Confirm password')"
+                />
+                <span class="icon is-small is-left">
+                  <fa-icon icon="lock" />
+                </span>
+              </p>
+            </div>
+          </div>
+          <div class="sgnup-buttons">
+            <div>
+              <p class="control has-text-centered">
+                <button type="submit" class="button is-login">
+                  {{ $gettext("Sign up") }}
+                </button>
+              </p>
+            </div>
+          </div>
+          <div>
+            <p class="has-text-danger has-text-centered" v-if="fail">
+              {{ fail }}
+            </p>
+          </div>
+          <div class="signup-link">
+            <button @click="$router.push({ name: 'Login' })" type="button">
+              {{ $gettext("Already have an account ?") }}
+            </button>
+          </div>
+        </form>
+      </div>
+    </div>
+    <router-view></router-view>
+  </section>
+</template>
+
+<script lang="ts">
+  import { Options, Vue } from "vue-class-component"
+  @Options({
+    name: "Signup",
+    data() {
+      return {
+        email: "",
+        lastName: "",
+        firstName: "",
+        password: "",
+        confirmPassword: "",
+        fail: "",
+      }
+    },
+    methods: {
+      async submit(): Promise<void> {
+        try {
+          this.$loading.show()
+          await this.$store.dispatch("signup", {
+            login: this.email,
+            lastname: this.lastName,
+            firstname: this.firstName,
+            password: this.password,
+            confirm_password: this.confirmPassword,
+          })
+          this.$router.push({ name: "Login" })
+          this.$msg.success(
+            this.$gettext("Your account has been created successfully.")
+          )
+        } catch (e) {
+          this.fail =
+            this.$gettext(
+              "Unexpected issue when attempting to connect to remote server."
+            ) +
+            "(" +
+            e +
```

Comment:
hmm, this will display un translated text, and probably barbaric one ! It should probably not be displayed to the user. Were you thinking about a particular error message ?